internal/server/iigointernal/judiciary.go

```diff
@@ -118,7 +140,96 @@ func (j *judiciary) appointNextPresident(currentPresident shared.ClientID, allIs
 	return nextPresident
 }
 
-// cycleSanctionCache rolls the sanction cache one turn forward (effectively dropping any sanctions longer than the depth)
+// updateSanctionScore uses results of InspectHistory to assign sanction scores to clients
+// This function relies upon (in part) the client provided RuleViolationSeverity
+func (j *judiciary) updateSanctionScore() {
+	//Clearing sanction map
+	j.sanctionRecord = map[shared.ClientID]roles.IIGOSanctionScore{}
+	islandTransgressions := map[shared.ClientID][]string{}
+	for k, v := range j.evaluationResults {
+		islandTransgressions[k] = unpackSingleIslandTransgressions(v)
+	}
+	j.scoreIslandTransgressions(islandTransgressions)
+}
+
+// scoreIslandTransgressions uses the client provided ruleViolationSeverity map to score island transgressions
+func (j *judiciary) scoreIslandTransgressions(transgressions map[shared.ClientID][]string) {
+	for islandId, rulesBroken := range transgressions {
+		totalIslandTurnScore := roles.IIGOSanctionScore(0)
+		for _, ruleBroken := range rulesBroken {
+			if score, ok := j.ruleViolationSeverity[ruleBroken]; ok {
+				totalIslandTurnScore += score
+			} else {
+				totalIslandTurnScore += roles.IIGOSanctionScore(1)
+			}
+		}
+		j.sanctionRecord[islandId] += totalIslandTurnScore
+	}
+}
+
+// applySanctions uses RulesInPlay and it's versions of the sanction rules to work out how much to sanction an island
+func (j *judiciary) applySanctions() {
+	j.cycleSanctionCache()
+	var currentSanctions []roles.Sanction
+	for islandID, sanctionScore := range j.sanctionRecord {
+		islandSanctionTier := getIslandSanctionTier(sanctionScore, j.sanctionThresholds)
+		sanctionEntry := roles.Sanction{
+			ClientID:     islandID,
+			SanctionTier: islandSanctionTier,
+			TurnsLeft:    sanctionLength,
+		}
+		currentSanctions = append(currentSanctions, sanctionEntry)
+	}
+	j.localSanctionCache[0] = currentSanctions
+}
+
+// sanctionEvaluate allows the clients to effectively pardon islands, levy and communicate sanctions
+func (j *judiciary) sanctionEvaluate(reportedIslandResources map[shared.ClientID]shared.Resources) {
```

Comment:
can't follow what's written here. break down into smaller fns.

Reply:
+ test

Reply:
Fair enough, I've broken into smaller functions. In terms of testing, since this relies on the globalRulesCache and other elements such as broadcasting communications. Is it ok if we hold the testing until integration tests for IIGO as a whole.

Reply:
if you guarantee it works :)

Reply:
I will guarantee that it works in the context of this PR, and will make sure to thoroughly test in the post-MVP orchestration where it will be called.

Reply:
Ok it's tested now ;)